File: app/src/main/java/com/android/heaton/easyvote/VoteData.java
package com.android.heaton.easyvote;

/**
 * Created by heaton on 16/4/1.
 */
public class VoteData {
    public String voteCode = "vote code";
    public String title;
    public String authorName = "Heaton";
    public String authorCode = "heaton_id";
    public String authorIcon ="";
    public String voteImage = "";
    public int localImage;
    public long startTime;
    public long endTime;
    public String option1title = "Option 1 Option 1 Option 1 Option 1Option 1 Option 1Option 1 Option 1";
    public String option1code = "option 1 code";
    public int option1count = 0;

    public String option2title = "Option 2";
    public String option2code = "option 2 code";
    public int option2Count = 0;
    public String optionTopTitle = "Option Champion Option Champion Option Champion Option Champion Option Champion";
    public String optionTopCode = "NONE";
    public int optionTopCount = 2;
    public String optionUserChoiceTitle = "Option User";
    public String optionUserChoiceCode = "2";
    public int optionUserChoiceCount = 0;
    public int minOption = 1;
    public int maxOption = 1;
    public int optionCount = 5;
    public int pollCount;
    public boolean isPolled = false;
    public boolean isFavorite;

    // TODO: OPTION TYPE
    public String pollType;
}

File: funnyvote/app/src/main/java/com/heaton/funnyvote/data/local/entity/VoteData.kt
package com.heaton.funnyvote.data.local.entity

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

@Entity(tableName = "votes")
data class VoteData(
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0,

    @SerializedName("c")
    var voteCode: String = "",

    @SerializedName("t")
    var title: String? = null,

    @SerializedName("nn")
    var authorName: String? = null,

    @SerializedName("token")
    var authorCode: String? = null,

    @SerializedName("tokentype")
    var authorCodeType: String? = null,

    @SerializedName("mi")
    var authorIcon: String? = null,

    @SerializedName("i")
    var voteImage: String? = null,

    var localImage: Int = 0,

    @SerializedName("on")
    var startTime: Long = 0,

    @SerializedName("off")
    var endTime: Long = 0,

    @SerializedName("min")
    var minOption: Int = 0,

    @SerializedName("max")
    var maxOption: Int = 0,

    @SerializedName("osn")
    var optionCount: Int = 0,

    @SerializedName("voted")
    var pollCount: Int = 0,

    @SerializedName("isVoted")
    var isPolled: Boolean = false,

    @SerializedName("fav")
    var isFavorite: Boolean = false,

    @SerializedName("res")
    var isCanPreviewResult: Boolean = false,

    @SerializedName("add")
    var isUserCanAddOption: Boolean = false,

    @SerializedName("p")
    var isNeedPassword: Boolean = false,

    @SerializedName("sec")
    var security: String = SECURITY_PUBLIC,

    var category: String? = null,
    var displayOrder: Int = 0,
    var pollType: String? = null,

    // Extracted option fields for local DB ease, as original did
    var option1Title: String? = null,
    var option1Code: String? = null,
    var option1Count: Int = 0,
    var option1Polled: Boolean = false,

    var option2Title: String? = null,
    var option2Code: String? = null,
    var option2Count: Int = 0,
    var option2Polled: Boolean = false,

    var optionTopTitle: String? = null,
    var optionTopCode: String? = null,
    var optionTopCount: Int = 0,
    var optionTopPolled: Boolean = false,

    var optionUserChoiceTitle: String? = null,
    var optionUserChoiceCode: String? = null,
    var optionUserChoiceCount: Int = 0
) {
    @Ignore
    @SerializedName("os")
    var netOptions: List<Option>? = null

    companion object {
        const val SECURITY_PRIVATE = "01"
        const val SECURITY_PUBLIC = "00"
        const val CATEGORY_HOT = "hot"
    }

    val isMultiChoice: Boolean
        get() = !(maxOption == 1 && minOption == 1)
}
